app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -338,6 +338,28 @@ class BrowserRobot {
         assertTrue(pausedStateMessage.waitForExists(waitingTime))
     }
 
+    fun swipeNavBarRight(tabUrl: String) {
+        // failing to swipe on Firebase sometimes, so it tries again
+        try {
+            navURLBar().perform(ViewActions.swipeRight())
+            assertTrue(mDevice.findObject(UiSelector().text(tabUrl)).waitUntilGone(waitingTime))
+        } catch (e: AssertionError) {
+            navURLBar().perform(ViewActions.swipeRight())
```

Comment:
Why is the action repeated in the catch?

Same below

Reply:
I tried to make it repeat the swipe in case the first time didn't work, without failing (twice I think is enough). Or do you think there's a better way to do that, with a loop? You can see the first runs of this test, without this hack that the swipe didn't do anything: https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/7088753260638706015/executions/bs.80bdb045aa9a45ce/videos